src/app.controller.ts

```diff
@@ -631,24 +653,46 @@ export class AppController {
       throw new BadRequestException("user not found!");
 
     const userWithEmail = await this.userService.findByEmail(body.email);
-    if(userWithEmail && (userWithEmail.userid.toString() != user.userid.toString()))
-      throw new BadRequestException("This E-Mail is being used by another user!");
+    if (
+      userWithEmail &&
+      userWithEmail.userid.toString() != user.userid.toString()
+    ) {
+      await this.userService.merge(req.user.userid, userWithEmail.userid);
```

Comment:
@EeroHeikkinen isn't is this dangerous thing to do.
Now any user can somebody else's e-mail and merge its everything with his own? Am I missing something here? Same thing is valid for the phone number below.

Reply:
I cannot access the logs at the moment but I recall this was previously discussed in the Whatsapp group, and the intention I got from Ma was that we should for now trust the users inputting whichever emails they provide as their own when filling the form on the platform.

The proper way to avoid people stealing points belonging to some other email would be to require an email verification step for the email merge, or an OTP code in the case of phone numbers. But that was deemed as unnecessarily complicated at this point. 

  For now also the fact that they need to use a social media account to attempt this attack could be reducing the motivation a little (since we can see the social id they used to do the attack, unless they went so far to actually create a fake account just for this purpose).

However I think we should definitely keep the email verification/OTP option open as a future step in case we start running into reports of people abusing the hidden opportunity to merge their points with some other user if they have  their personal data.

Perhaps most crucially we would need to address this concern at the potential point in the future where we would implement some ways to redeem the points for actual real world benefits instead of just virtual bragging rights :)

Reply:
I definitely agree with you. However, I am still not confident about merging with this new feature. This feature introduces a form that a user can freely change his e-mail address. Before then there was no form and we can trust the calling 3rd party API which is fine.  But when introducing the form to the user, I am also not expecting a malicious behavior however with this state of the change it is very easy to an un-intentionally merging accounts that should not be merged. Hence the user is not aware of his action (or he is not informed) that he is actually may merge accounts. 

1-I am still questioning what benefit we have by merging accounts without questioning (OTP/email verification etc...) by default. I think we can start merging accounts if we observe too many duplicate accounts for single actual user. and then we can create feature that both OTP code/email verification secure and make the merge calls. 

2-If we are going to merge them by default then before approving this branch I think we should at least add a feature to this branch that if the user enters an existing e-mail address and/or phone number a warning message appears and informs the user that he is entering an existing account info and if he continues the accounts will be merged, so that at least we sould make sure the action will be a conscious action.

what do you think?

Reply:
Thanks for the excellent points!

1. I understood the main case to be related to situation if a user has an old social media account with different email than their current primary email (I have situation like this with my 10+ year old FB account). So if they register as earth buddy with email A, and they use their social profile to later login with email B - they would need to provide the email in the form and merge to access the points with their primary account.

2. I agree confirming via prompt would be the best way to go. Great idea! Can you figure out the best way to access the message? (Quickly thinking of options like a separate endpoint to check if the email/phone exists before submitting the main form via POST, or to return a different output in the POST submit of the form in case email/phone exists, and to require setting a confirm parameter if the email exists)